src/error.rs

```diff
@@ -180,12 +180,36 @@ pub enum LimitErrorKind {
     /// The specified strict limits are not supported for this operation
     Unsupported {
         /// The given limits
-        limits: crate::Limits,
+        limits: LimitsSnapshot,
```

Comment:
The only place that `LimitsSnapshot` is used is this never-constructed variant of `LimitErrorKind`. That doesn't feel like it is carrying its weight to have as another type listed in our public API...